modules/heat_transfer/src/physics/HeatConductionFV.C

```diff
@@ -0,0 +1,164 @@
+
+  return params;
+}
+
+HeatConductionFV::HeatConductionFV(const InputParameters & parameters)
+  : HeatConductionPhysics(parameters)
+{
+}
+
+void
+HeatConductionFV::initializePhysicsAdditional()
+{
+  getProblem().needFV();
+}
+
+void
+HeatConductionFV::addFVKernels()
+{
+  {
+    const std::string kernel_type = "FVDiffusion";
+    InputParameters params = getFactory().getValidParams(kernel_type);
+    params.set<NonlinearVariableName>("variable") = _temperature_name;
+    params.set<MooseFunctorName>("coeff") =
+        getParam<MooseFunctorName>("thermal_conductivity_functor");
+    getProblem().addFVKernel(kernel_type, prefix() + _temperature_name + "_conduction", params);
+  }
+  if (isParamValid("heat_source_var"))
+  {
+    const std::string kernel_type = "FVCoupledForce";
+    InputParameters params = getFactory().getValidParams(kernel_type);
+    params.set<NonlinearVariableName>("variable") = _temperature_name;
+    params.set<MooseFunctorName>("v") = getParam<VariableName>("heat_source_var");
+    getProblem().addFVKernel(kernel_type, prefix() + _temperature_name + "_source", params);
+  }
+  if (isTransient())
+  {
+    const std::string kernel_type = "FVHeatConductionTimeDerivative";
+    InputParameters params = getFactory().getValidParams(kernel_type);
+    params.set<NonlinearVariableName>("variable") = _temperature_name;
+    params.applyParameter(parameters(), "specific_heat");
+    params.set<MaterialPropertyName>("density_name") = getParam<MaterialPropertyName>("density");
+    getProblem().addFVKernel(kernel_type, prefix() + _temperature_name + "_time", params);
+  }
+}
+
+void
+HeatConductionFV::addFVBCs()
+{
+  // We dont need to add anything for insulated boundaries, 0 flux is the default boundary condition
+  if (isParamValid("heat_flux_boundaries"))
+  {
+    const std::string bc_type = "FVFunctorNeumannBC";
+    InputParameters params = getFactory().getValidParams(bc_type);
+    params.set<NonlinearVariableName>("variable") = _temperature_name;
+
+    const auto & heat_flux_boundaries = getParam<std::vector<BoundaryName>>("heat_flux_boundaries");
+    const auto & boundary_heat_fluxes =
+        getParam<std::vector<MooseFunctorName>>("boundary_heat_fluxes");
+    // Optimization if all the same
+    if (std::set<MooseFunctorName>(boundary_heat_fluxes.begin(), boundary_heat_fluxes.end())
+                .size() == 1 &&
+        heat_flux_boundaries.size() > 1)
+    {
+      params.set<std::vector<BoundaryName>>("boundary") = heat_flux_boundaries;
+      params.set<MooseFunctorName>("functor") = boundary_heat_fluxes[0];
+      getProblem().addFVBC(bc_type, prefix() + _temperature_name + "_heat_flux_bc_all", params);
+    }
+    else
+    {
+      for (const auto i : index_range(heat_flux_boundaries))
+      {
+        params.set<std::vector<BoundaryName>>("boundary") = {heat_flux_boundaries[i]};
+        params.set<MooseFunctorName>("functor") = boundary_heat_fluxes[i];
+        getProblem().addFVBC(bc_type,
```

Comment:
This code works because we copy construct input parameters in the factory creation. This code makes me nervous, but I wouldn't recommend changing it

Reply:
yeah this practice of reusing the parameters is widly used already